Detections/CyberpionSecurityLogs/HighUrgencyActionItems.yaml

```diff
@@ -0,0 +1,38 @@
+id: 8e0403b1-07f8-4865-b2e9-74d1e83200a4
+name: High Urgency Cyberpion Action Items
+description: |
+  'This query creates an alert for active Cyberpion Action Items with high urgency (9-10).
+   Urgency can be altered using the "min_urgency" variable in the query.'
+severity: High
+requiredDataConnectors:
+  - connectorId: CyberpionSecurityLogs
+    dataTypes:
+      - CyberpionActionItems_CL
+queryFrequency: 1d
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1190
+  - T1195
+query: |
+  let timeframe = 14d;
+  let time_generated_bucket = 1h;
+  let min_urgency = 9;
+  let maxTimeGeneratedBucket = toscalar(
+     CyberpionActionItems_CL
+     | where TimeGenerated > ago(timeframe)
+     | summarize max(bin(TimeGenerated, time_generated_bucket))
+     );
+  CyberpionActionItems_CL
+   | where TimeGenerated > ago(timeframe) and is_open_b == true
+   | where bin(TimeGenerated, time_generated_bucket) == maxTimeGeneratedBucket
+   | where urgency_d >= min_urgency
+   | extend timestamp = opening_datetime_t
+entityMappings:
+  - entityType: DNS
```

Comment:
The entity has to be extracted from your query results and should generally be present in the last extend statement. Could you please check if that can be done here. Also, looking at the column name field "host_s" sounds like this should be a HostCustomEntity with a entity type of Host vs DNS. If that is the case would you please alter this section accordingly.
Here are some more details regarding this: https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#entitymappings for general guidance regarding this. 
Thanks.

Reply:
Hey, thanks for the reply.
host_s isn't a "Host, Computer, System, Device" in the connector. It is a domain or subdomain (any of the following - www.example.com / www.mail.example.com / example.com). 
Is there a field other than DomainName I can map it to?

Regarding the extension of the DNS mapping, I'd be happy to extend the host_s field name to be something like DNSCustomEntity just for readability. Is what you mean?

Thanks!

Reply:
Hello?

Reply:
Thanks Yotam for the explanation around the host_s field. That makes sense. If there is a link to a schema reference, we could probably also include that in the description section of the query. 

For the DNS Entity type the identifier is DomainName. 
Yes. If you could  please extend the host_s field name to DNSCustomEntity and have the entity mapping to somewhat like below that would be great. Thanks again.

- entityType: DNS
    fieldMappings:
      - identifier: DomainName
        columnName: DNSCustomEntity

Reply:
Done.
About the schema reference, we don't have one right now and I think that by extending host_s = DNSCustomEntity like you said, it becomes clear.

By the way, I think maybe you should add the new step of adding the Connector to "ValidConnectorIds.json" to the steps in the documentation (if that is what I should have done).

Thanks again!

Reply:
Thanks Yotam.

Reply:
So can I get an approval?